packages/x-components/src/components/base-teleport.vue

```diff
@@ -16,12 +16,39 @@ export default defineComponent({
       type: String,
       required: true,
     },
+    hideSiblings: {
+      type: Boolean,
+      default: true,
+    },
+    position: {
+      type: Number,
+      required: false,
+    },
+  },
+  setup(props) {
+    onMounted(() => {
+      const parentElement = document.querySelector(props.target)
+      const teleportElement = parentElement?.querySelector('.x-base-teleport')
+
+      if (!props.hideSiblings) {
+        if (parentElement && teleportElement) {
+          if (props.position && props.position < parentElement.children.length) {
+            parentElement.insertBefore(teleportElement, parentElement.children[props.position])
+          } else {
+            // Add as the last child
+            parentElement.appendChild(teleportElement)
+          }
+        }
+      } else {
+        teleportElement?.classList.add('x-base-teleport__hide-siblings')
+      }
+    })
   },
 })
 </script>
 
 <style lang="css">
-:has(> .x-base-teleport) > *:not(.x-base-teleport) {
+:has(> .x-base-teleport__hide-siblings) > *:not(.x-base-teleport) {
```

Comment:
un `BEM` `__` means `element`, but `hide-siblings` looks like a  `variant` or `state`

```suggestion
:has(> .x-base-teleport--hide-siblings) > *:not(.x-base-teleport) {
```